client/extensions/woocommerce/products/product-variation-types-form/index.js

```diff
@@ -0,0 +1,120 @@
+
+export default class ProductVariationTypesForm extends Component {
+
+	static propTypes = {
+		product: PropTypes.shape( {
+			id: PropTypes.number.isRequired,
+			name: PropTypes.string.isRequired,
+			type: PropTypes.string.isRequired,
+		} ),
+		variations: PropTypes.arrayOf( PropTypes.shape( {
+			type: PropTypes.string.isRequired,
+			values: PropTypes.arrayOf( PropTypes.string )
+		} ) ),
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			variations: this.props.variations ? this.props.variations : this.getInitialFields(),
+		};
+
+		this.addVariation = this.addVariation.bind( this );
+		this.renderInputs = this.renderInputs.bind( this );
+	}
+
+	getInitialFields() {
+		return [ this.getNewFields() ];
+	}
+
+	getNewFields() {
+		return {
+			type: '',
+			values: [],
+		};
+	}
+
+	updateType( index, event ) {
+		event.preventDefault();
+		const newValue = event.target.value,
+			updatedVariations = clone( this.state.variations );
+		updatedVariations[ index ] = clone( updatedVariations[ index ] );
+		updatedVariations[ index ].type = newValue;
+		this.setState( { variations: updatedVariations } );
+	}
+
+	updateValues( index, value ) {
+		const updatedVariations = clone( this.state.variations );
+		updatedVariations[ index ] = clone( updatedVariations[ index ] );
+		updatedVariations[ index ].values = value;
+		this.setState( { variations: updatedVariations } );
+	}
+
+	renderInputs( variation, index ) {
+		return (
+			<div key={index} className="product-variation-types-form__fieldset">
+				<FormTextInput
+					placeholder={ i18n.translate( 'Color' ) }
+					value={ variation.type }
+					name="type"
+					onChange={ this.updateType.bind( this, index ) }
+					className="product-variation-types-form__field"
+				/>
+				<TokenField
+					placeholder={ i18n.translate( 'Comma separate these' ) }
+					value={ variation.values }
+					name="values"
+					onChange={ this.updateValues.bind( this, index ) }
+				/>
+			</div>
+		);
+	}
+
+	addVariation( event ) {
+		event.preventDefault();
+		const updatedVariations = this.state.variations.concat( [ this.getNewFields() ] );
+		this.setState( { variations: updatedVariations } );
+	}
+
+	render() {
+		const fields = this.state.variations,
+			inputs = fields.map( this.renderInputs );
+		return (
+			<div className="product-variation-types-form__wrapper">
+				<strong>{ i18n.translate( 'Okay! Let\'s add some variations.' ) }</strong>
+				<p>
+					{ i18n.translate(
+						'A common {{em}}variation type{{/em}} is color. The {{em}}values{{/em}} would be the colors the product is available in.',
+						{ components: { em: <em /> } }
+					) }
+				</p>
+
+				<div className="product-variation-types-form__group">
+					<div className="product-variation-types-form__labels">
+						<FormLabel className="product-variation-types-form__label">{ i18n.translate( 'Variation type' ) }</FormLabel>
+						<FormLabel>{ i18n.translate( 'Values' ) }</FormLabel>
```

Comment:
Hi! I've found a possible matching string that has already been translated **90** times:
`translate( 'Value' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).